sdk/identity/azure_identity/src/lib.rs

```diff
@@ -112,26 +112,75 @@ fn get_authority_host(env: Option<Env>, option: Option<String>) -> Result<Url> {
+    const FRAGMENT: &'static str;
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl TroubleshootingGuide for AzureCliCredential {
+    const FRAGMENT: &'static str = "#azure-cli";
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl TroubleshootingGuide for AzureDeveloperCliCredential {
+    const FRAGMENT: &'static str = "#azd";
+}
+
+impl TroubleshootingGuide for AzurePipelinesCredential {
+    const FRAGMENT: &'static str = "#apc";
+}
+
+#[cfg(feature = "client_certificate")]
+impl TroubleshootingGuide for ClientCertificateCredential {
+    const FRAGMENT: &'static str = "#client-cert";
+}
+
+impl TroubleshootingGuide for ClientSecretCredential {
+    const FRAGMENT: &'static str = "#client-secret";
+}
+
+impl TroubleshootingGuide for ManagedIdentityCredential {
+    const FRAGMENT: &'static str = "#managed-id";
+}
+
+impl TroubleshootingGuide for WorkloadIdentityCredential {
+    const FRAGMENT: &'static str = "#workload";
+}
+
+mod private {
+    pub trait Sealed {}
+    #[cfg(not(target_arch = "wasm32"))]
+    impl Sealed for super::AzureCliCredential {}
+    #[cfg(not(target_arch = "wasm32"))]
+    impl Sealed for super::AzureDeveloperCliCredential {}
+    impl Sealed for super::AzurePipelinesCredential {}
+    #[cfg(feature = "client_certificate")]
+    impl Sealed for super::ClientCertificateCredential {}
+    impl Sealed for super::ClientSecretCredential {}
+    impl Sealed for super::ManagedIdentityCredential {}
+    impl Sealed for super::WorkloadIdentityCredential {}
+}
 
 /// Map an error from a credential's get_token() method to an ErrorKind::Credential error, appending
 /// a link to the troubleshooting guide entry for that credential, if it has one.
 ///
 /// TODO: decide whether to map to ErrorKind::Credential here (https://github.com/Azure/azure-sdk-for-rust/issues/3127)
-fn authentication_error<T: 'static>(e: azure_core::Error) -> azure_core::Error {
+fn authentication_error<T: TroubleshootingGuide + 'static>(
+    e: azure_core::Error,
+) -> azure_core::Error {
     azure_core::Error::with_message_fn(e.kind().clone(), || {
         let type_name = std::any::type_name::<T>();
-        let short_name = type_name.rsplit("::").next().unwrap_or(type_name); // cspell:ignore rsplit
-        let link = match short_name {
-            "AzureCliCredential" => format!("{TSG_LINK}azd"),
-            "AzureDeveloperCliCredential" => format!("{TSG_LINK}azure-cli"),
-            "AzurePipelinesCredential" => format!("{TSG_LINK}apc"),
-            #[cfg(feature = "client_certificate")]
-            "ClientCertificateCredential" => format!("{TSG_LINK}client-cert"),
-            "ClientSecretCredential" => format!("{TSG_LINK}client-secret"),
-            "ManagedIdentityCredential" => format!("{TSG_LINK}managed-id"),
-            "WorkloadIdentityCredential" => format!("{TSG_LINK}workload"),
-            _ => "".to_string(),
+        let short_name = type_name.rsplit("::").next().unwrap_or(type_name);
```

Comment:
Oh, shoot. I didn't realize originally you *also* needed the `short_name` for the message below. In that case, what you had was fine (though I'd still put the anchor in the format expression). I'll leave it up to you if you want to revert or not. Sorry I missed that.